cip-smart-tx-signing/README.md

```diff
@@ -0,0 +1,80 @@
+---
+CIP: ?
+Title: Wallet Transaction Caching
+Status: Proposed
+Category: Wallets
+Authors:
+    - Josh Marchand <josh@securitybot.info>
+Implementors: []
+Discussions:
+ - https://github.com/cardano-foundation/CIPs/pull/733
+Created: 2023-12-22
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This CIP extends [CIP-0030 Cardano dApp-Wallet Web Bridge](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030/) and [CIP-0103? | Web-Wallet Bridge - Bulk Transaction Signing](https://github.com/cardano-foundation/CIPs/pull/587) to allow wallets to provide a complete and accurate summary of transactions that spend outputs that don't exist on chain yet when possible, and clearly communicate to the user risks when they cannot.
+
+## Motivation: why is this CIP necessary?
+
+Transaction chaining, building transactions that spend outputs of transactions that aren't settled on chain, allows dApps to quickly do several on-chain transactions instead of waiting for each indvidual transaction to settle before building and submitting the next one. This is a powerful feature that has allowed dApps to provide seamless experiences to users. However, due to the tranasaction data structure, wallets are unable to provide a complete and accurate summary of these chained transactions when prompting users for a signature.
```

Comment:
```suggestion
Transaction chaining, building transactions that spend outputs of transactions that aren't settled on chain, allows dApps to quickly do several on-chain transactions instead of waiting for each individual transaction to settle before building and submitting the next one. This is a powerful feature that has allowed dApps to provide seamless experiences to users. However, due to the transactions data structure, wallets are unable to provide a complete and accurate summary of these chained transactions when prompting users for a signature.
```

spelling